be/src/formats/csv/array_reader.h

```diff
@@ -0,0 +1,42 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2022-present, StarRocks Limited.
+
+#pragma once
+
+#include "formats/csv/converter.h"
+
+namespace starrocks::vectorized::csv {
+class ArrayReader {
+public:
+    explicit ArrayReader(char array_delimiter) : _array_delimiter(array_delimiter) {}
+    virtual ~ArrayReader() = default;
+    [[nodiscard]] virtual bool read_string(const std::unique_ptr<Converter>& elem_converter, Column* column, Slice s,
+                                           const Converter::Options& options) const = 0;
+    [[nodiscard]] virtual bool split_array_elements(Slice s, std::vector<Slice>* elements) const = 0;
+
+protected:
+    char _array_delimiter;
+};
+
+// Reader for standard array format: [[1, 2], [3, 4, 5]].
+class StandardArrayReader final : public ArrayReader {
+public:
+    explicit StandardArrayReader() : ArrayReader(',') {}
+    [[nodiscard]] bool read_string(const std::unique_ptr<Converter>& elem_converter, Column* column, Slice s,
+                                   const Converter::Options& options) const override;
+    [[nodiscard]] bool split_array_elements(Slice s, std::vector<Slice>* elements) const override;
+};
+
+// Reader for Hive text array format.
```

Comment:
Please add some comments to explain what collection delimiter and mapkey delimiter do